network/test/epochtransition_test.go

```diff
@@ -175,38 +280,125 @@ func checkConnectivity(t *testing.T, mw *p2p.Middleware, ids flow.IdentityList)
 	}, 5*time.Second, time.Millisecond*100)
 }
 
-// sendMessagesAndVerify sends a message from each engine to the other engines
-// and verifies that all the messages are delivered.
-func sendMessagesAndVerify(t *testing.T, ids flow.IdentityList, engs []*MeshEngine, send ConduitSendWrapperFunc) {
+// assertDisconnected checks that the middleware of a node is not connected to any of the other nodes specified in the
+// ids list
+func (suite *MutableIdentityTableSuite) assertDisconnected(mw *p2p.Middleware, ids flow.IdentityList) {
+	t := suite.T()
+	require.Eventually(t, func() bool {
+		for _, id := range ids {
+			connected, err := mw.IsConnected(*id)
+			require.NoError(t, err)
+			if connected {
+				return false
+			}
+		}
+		return true
+	}, 5*time.Second, time.Millisecond*100)
+}
+
+func (suite *MutableIdentityTableSuite) assertNetworkPrimitives(
+	allowedIDs flow.IdentityList,
+	allowedEngs []*MeshEngine,
+	disallowedIDs flow.IdentityList,
+	disallowedEngs []*MeshEngine) {
+	suite.Run("Publish", func() {
+		suite.exchangeMessages(allowedIDs, allowedEngs, disallowedIDs, disallowedEngs, suite.Publish, false)
```

Comment:
`publish` and `multicast` Send won't fail even if the targets are non-existent or evicted, hence passing in `false` for  `expectSendErrorForDisallowedIDs`